diff --git a/osad.spec b/osad.spec
index 70eb1b150e..3e726ab1aa 100644
--- a/osad.spec
+++ b/osad.spec
@@ -18,13 +18,17 @@
 %global default_py3 1
 %endif
 
+%if ( 0%{?fedora} && 0%{?fedora} < 28 ) || ( 0%{?rhel} && 0%{?rhel} < 8 )
+%global build_py2   1
+%endif
+
 %define pythonX %{?default_py3: python3}%{!?default_py3: python2}
 
 Name: osad
 Summary: Open Source Architecture Daemon
 License: GPLv2
 Version: 5.11.99
-Release: 3%{?dist}
+Release: 4%{?dist}
 URL:     https://github.com/spacewalkproject/spacewalk
 Source0: https://github.com/spacewalkproject/spacewalk/archive/%{name}-%{version}.tar.gz
 BuildArch: noarch
@@ -73,6 +77,7 @@ commands are instantly executed.
 This package effectively replaces the behavior of rhnsd/rhn_check that
 only poll the Spacewalk Server from time to time.
 
+%if 0%{?build_py2}
 %package -n python2-%{name}
 Summary: Open Source Architecture Daemon
 %{?python_provide:%python_provide python2-%{name}}
@@ -89,6 +94,7 @@ Requires: python-hashlib
 BuildRequires: python-devel
 %description -n python2-%{name}
 Python 2 specific files for %{name}
+%endif
 
 %if 0%{?build_py3}
 %package -n python3-%{name}
@@ -156,8 +162,13 @@ that message is transported via jabber protocol to OSAD agent on the clients.
 
 %package -n python2-osa-dispatcher
 Summary: OSA dispatcher
+%if 0%{?fedora} >= 28
+BuildRequires: python2-devel
+Requires: python2
+%else
 BuildRequires: python-devel
 Requires: python
+%endif
 Requires: jabberpy
 Requires: python2-osa-common = %{version}
 %description -n python2-osa-dispatcher
@@ -282,6 +293,11 @@ install -d %{buildroot}%{_sbindir}
 install -p -m 755 osa-dispatcher-selinux/osa-dispatcher-selinux-enable %{buildroot}%{_sbindir}/osa-dispatcher-selinux-enable
 %endif
 
+%if ! 0%{?build_py2}
+rm -rf $RPM_BUILD_ROOT/%{python_sitelib}/osad/osad*
+rm -f $RPM_BUILD_ROOT/usr/sbin/osad-%{python_version}
+%endif
+
 %clean
 
 %{!?systemd_post: %global systemd_post() if [ $1 -eq 1 ] ; then /usr/bin/systemctl enable %%{?*} >/dev/null 2>&1 || : ; fi; }
@@ -427,12 +443,14 @@ rpm -ql osa-dispatcher | xargs -n 1 /sbin/restorecon -rvi {}
 %dir %{_sysconfdir}/sysconfig/rhn/clientCaps.d
 %endif
 
+%if 0%{?build_py2}
 %files -n python2-%{name}
 %attr(755,root,root) %{_sbindir}/osad-%{python_version}
 %dir %{python_sitelib}/osad
 %{python_sitelib}/osad/osad.py*
 %{python_sitelib}/osad/osad_client.py*
 %{python_sitelib}/osad/osad_config.py*
+%endif
 
 %if 0%{?build_py3}
 %files -n python3-%{name}
@@ -514,6 +532,14 @@ rpm -ql osa-dispatcher | xargs -n 1 /sbin/restorecon -rvi {}
 %endif
 
 %changelog
+* Tue Mar 20 2018 Tomas Kasparek <tkasparek@redhat.com> 5.11.99-4
+- remove osad files when packaging only for python3 (tkasparek@redhat.com)
+- osa-dispatcher is dependent on spacewalk-backend which is in python2
+  (tkasparek@redhat.com)
+- run osa-dispatcher on python3 when possible (tkasparek@redhat.com)
+- don't build python2 subpackages on F28 + update python requires
+  (tkasparek@redhat.com)
+
 * Wed Mar 14 2018 Tomas Kasparek <tkasparek@redhat.com> 5.11.99-3
 - build osad for python3 (tkasparek@redhat.com)
 
