diff --git a/osad.spec b/osad.spec
index 3e726ab1aa..2e6884cf75 100644
--- a/osad.spec
+++ b/osad.spec
@@ -28,7 +28,7 @@ Name: osad
 Summary: Open Source Architecture Daemon
 License: GPLv2
 Version: 5.11.99
-Release: 4%{?dist}
+Release: 5%{?dist}
 URL:     https://github.com/spacewalkproject/spacewalk
 Source0: https://github.com/spacewalkproject/spacewalk/archive/%{name}-%{version}.tar.gz
 BuildArch: noarch
@@ -77,25 +77,6 @@ commands are instantly executed.
 This package effectively replaces the behavior of rhnsd/rhn_check that
 only poll the Spacewalk Server from time to time.
 
-%if 0%{?build_py2}
-%package -n python2-%{name}
-Summary: Open Source Architecture Daemon
-%{?python_provide:%python_provide python2-%{name}}
-Requires: %{name} = %{version}-%{release}
-Requires: python
-Requires: rhnlib >= 2.8.3
-Requires: spacewalk-usix
-Requires: jabberpy
-Requires: python2-rhn-client-tools >= 2.8.4
-Requires: python2-osa-common = %{version}
-%if 0%{?rhel} && 0%{?rhel} <= 5
-Requires: python-hashlib
-%endif
-BuildRequires: python-devel
-%description -n python2-%{name}
-Python 2 specific files for %{name}
-%endif
-
 %if 0%{?build_py3}
 %package -n python3-%{name}
 Summary: Open Source Architecture Daemon
@@ -112,16 +93,6 @@ BuildRequires: python3-devel
 Python 3 specific files for %{name}
 %endif
 
-%package -n python2-osa-common
-Summary: OSA common files
-Requires: jabberpy
-Conflicts: %{name} < %{version}-%{release}
-Conflicts: %{name} > %{version}-%{release}
-Obsoletes: osa-common <= 5.11.91
-Provides:  osa-common = %{version}
-%description -n python2-osa-common
-Python 2 common files needed by osad and osa-dispatcher
-
 %if 0%{?build_py3}
 %package -n python3-osa-common
 Summary: OSA common files
@@ -134,84 +105,6 @@ Provides:  osa-common = %{version}
 Python 3 common files needed by osad and osa-dispatcher
 %endif
 
-%package -n osa-dispatcher
-Summary: OSA dispatcher
-Requires: spacewalk-backend-server >= 1.2.32
-Requires: python2-osa-dispatcher = %{version}-%{release}
-Requires: lsof
-Conflicts: %{name} < %{version}-%{release}
-Conflicts: %{name} > %{version}-%{release}
-%if 0%{?suse_version} >= 1210
-%{?systemd_requires}
-%endif
-%if 0%{?suse_version}
-# provides chkconfig on SUSE
-Requires(post): aaa_base
-Requires(preun): aaa_base
-%else
-Requires(post): chkconfig
-Requires(preun): chkconfig
-# This is for /sbin/service
-Requires(preun): initscripts
-%endif
-
-%description -n osa-dispatcher
-OSA dispatcher is supposed to run on the Spacewalk server. It gets information
-from the Spacewalk server that some command needs to be execute on the client;
-that message is transported via jabber protocol to OSAD agent on the clients.
-
-%package -n python2-osa-dispatcher
-Summary: OSA dispatcher
-%if 0%{?fedora} >= 28
-BuildRequires: python2-devel
-Requires: python2
-%else
-BuildRequires: python-devel
-Requires: python
-%endif
-Requires: jabberpy
-Requires: python2-osa-common = %{version}
-%description -n python2-osa-dispatcher
-Python 2 specific files for osa-dispatcher.
-
-%if 0%{?build_py3}
-%package -n python3-osa-dispatcher
-Summary: OSA dispatcher
-BuildRequires: python3-devel
-Requires: python3
-Requires: python3-jabberpy
-Requires: python3-osa-common = %{version}
-%description -n python3-osa-dispatcher
-Python 3 specific files for osa-dispatcher.
-%endif
-
-%if 0%{?include_selinux_package}
-%package -n osa-dispatcher-selinux
-%global selinux_variants mls strict targeted
-%global selinux_policyver %(sed -e 's,.*selinux-policy-\\([^/]*\\)/.*,\\1,' /usr/share/selinux/devel/policyhelp 2> /dev/null)
-%global POLICYCOREUTILSVER 1.33.12-1
-
-%global moduletype apps
-%global modulename osa-dispatcher
-
-Summary: SELinux policy module supporting osa-dispatcher
-BuildRequires: checkpolicy, selinux-policy-devel, hardlink
-BuildRequires: policycoreutils >= %{POLICYCOREUTILSVER}
-Requires: spacewalk-selinux
-
-%if "%{selinux_policyver}" != ""
-Requires: selinux-policy >= %{selinux_policyver}
-%endif
-%if 0%{?rhel} == 5
-Requires:        selinux-policy >= 2.4.6-114
-%endif
-Requires(post): /usr/sbin/semodule, %{sbinpath}/restorecon, /usr/sbin/selinuxenabled, /usr/sbin/semanage
-Requires(postun): /usr/sbin/semodule, %{sbinpath}/restorecon, /usr/sbin/semanage, spacewalk-selinux
-Requires: osa-dispatcher
-
-%description -n osa-dispatcher-selinux
-SELinux policy module supporting osa-dispatcher.
-%endif
 
 %prep
 %setup -q
@@ -225,22 +118,6 @@ sed -i 's@^#!/usr/bin/python$@#!/usr/bin/python -s@' invocation.py
 %build
 make -f Makefile.osad all PYTHONPATH=%{python_sitelib}
 
-%if 0%{?include_selinux_package}
-%{__perl} -i -pe 'BEGIN { $VER = join ".", grep /^\d+$/, split /\./, "%{version}.%{release}"; } s!\@\@VERSION\@\@!$VER!g;' osa-dispatcher-selinux/%{modulename}.te
-%if 0%{?fedora} || 0%{?rhel} >= 7
-cat osa-dispatcher-selinux/%{modulename}.te.fedora17 >> osa-dispatcher-selinux/%{modulename}.te
-%endif
-%if 0%{?fedora} >= 26
-cat osa-dispatcher-selinux/%{modulename}.te.fedora26 >> osa-dispatcher-selinux/%{modulename}.te
-%endif
-for selinuxvariant in %{selinux_variants}
-do
-    make -C osa-dispatcher-selinux NAME=${selinuxvariant} -f /usr/share/selinux/devel/Makefile
-    mv osa-dispatcher-selinux/%{modulename}.pp osa-dispatcher-selinux/%{modulename}.pp.${selinuxvariant}
-    make -C osa-dispatcher-selinux NAME=${selinuxvariant} -f /usr/share/selinux/devel/Makefile clean
-done
-%endif
-
 %install
 install -d $RPM_BUILD_ROOT%{rhnroot}
 make -f Makefile.osad install PREFIX=$RPM_BUILD_ROOT ROOT=%{rhnroot} INITDIR=%{_initrddir} \
@@ -272,31 +149,17 @@ install -m 0644 osad.service $RPM_BUILD_ROOT/%{_unitdir}/
 install -m 0644 osa-dispatcher.service $RPM_BUILD_ROOT/%{_unitdir}/
 %endif
 
-%if 0%{?include_selinux_package}
-for selinuxvariant in %{selinux_variants}
-  do
-    install -d %{buildroot}%{_datadir}/selinux/${selinuxvariant}
-    install -p -m 644 osa-dispatcher-selinux/%{modulename}.pp.${selinuxvariant} \
-           %{buildroot}%{_datadir}/selinux/${selinuxvariant}/%{modulename}.pp
-  done
-
-# Install SELinux interfaces
-install -d %{buildroot}%{_datadir}/selinux/devel/include/%{moduletype}
-install -p -m 644 osa-dispatcher-selinux/%{modulename}.if \
-  %{buildroot}%{_datadir}/selinux/devel/include/%{moduletype}/%{modulename}.if
-
-# Hardlink identical policy module packages together
-/usr/sbin/hardlink -cv %{buildroot}%{_datadir}/selinux
-
-# Install osa-dispatcher-selinux-enable which will be called in %%post
-install -d %{buildroot}%{_sbindir}
-install -p -m 755 osa-dispatcher-selinux/osa-dispatcher-selinux-enable %{buildroot}%{_sbindir}/osa-dispatcher-selinux-enable
-%endif
-
-%if ! 0%{?build_py2}
-rm -rf $RPM_BUILD_ROOT/%{python_sitelib}/osad/osad*
+rm -rf $RPM_BUILD_ROOT/%{python_sitelib}/*
 rm -f $RPM_BUILD_ROOT/usr/sbin/osad-%{python_version}
-%endif
+rm -f $RPM_BUILD_ROOT/usr/sbin/osa-dispatcher*
+rm -f $RPM_BUILD_ROOT/etc/logrotate.d/osa-dispatcher
+rm -f $RPM_BUILD_ROOT/etc/rhn/tns_admin/osa-dispatcher/sqlnet.ora
+rm -f $RPM_BUILD_ROOT/etc/sysconfig/osa-dispatcher
+rm -f $RPM_BUILD_ROOT/usr/share/rhn/config-defaults/rhn_osa-dispatcher.conf
+rm -f $RPM_BUILD_ROOT/var/log/rhn/osa-dispatcher.log
+rm -f $RPM_BUILD_ROOT/usr/lib/systemd/system/osa-dispatcher.service
+rm -f $RPM_BUILD_ROOT/%{python3_sitelib}/osad/dispatcher_client.py
+rm -f $RPM_BUILD_ROOT/%{python3_sitelib}/osad/osa_dispatcher.py
 
 %clean
 
@@ -352,78 +215,6 @@ fi
 %endif
 %endif
 
-%if 0%{?suse_version} >= 1210
-%pre
-%service_add_pre osad.service
-
-%pre -n osa-dispatcher
-%service_add_pre osa-dispatcher.service
-
-%postun -n osa-dispatcher
-%service_del_postun osa-dispatcher.service
-
-%endif
-
-%post -n osa-dispatcher
-%if 0%{?suse_version} >= 1210
-%service_add_post osa-dispatcher.service
-%else
-if [ -f %{_sysconfdir}/init.d/osa-dispatcher ]; then
-    /sbin/chkconfig --add osa-dispatcher
-fi
-if [ -f %{_unitdir}/osa-dispatcher.service ]; then
-    %systemd_post osa-dispatcher.service
-    if [ "$1" = "2" ]; then
-        # upgrade from old init.d
-        if [ -L /etc/rc2.d/S86osa-dispatcher ]; then
-            /usr/bin/systemctl enable osa-dispatcher.service >/dev/null 2>&1
-        fi
-        rm -f /etc/rc?.d/[SK]??osa-dispatcher
-    fi
-fi
-%endif
-
-%preun -n osa-dispatcher
-%if 0%{?suse_version} >= 1210
-%service_del_preun osa-dispatcher.service
-%else
-if [ $1 = 0 ]; then
-    %if 0%{?fedora} || 0%{?rhel} >= 7
-    %systemd_preun osa-dispatcher.service
-    %else
-    /sbin/service osa-dispatcher stop > /dev/null 2>&1
-    /sbin/chkconfig --del osa-dispatcher
-    %endif
-fi
-%endif
-
-%if 0%{?include_selinux_package}
-%post -n osa-dispatcher-selinux
-if /usr/sbin/selinuxenabled ; then
-   %{_sbindir}/osa-dispatcher-selinux-enable
-fi
-
-%posttrans -n osa-dispatcher-selinux
-#this may be safely remove when BZ 505066 is fixed
-if /usr/sbin/selinuxenabled ; then
-  rpm -ql osa-dispatcher | xargs -n 1 /sbin/restorecon -rvi {}
-  /sbin/restorecon -vvi /var/log/rhn/osa-dispatcher.log
-fi
-
-%postun -n osa-dispatcher-selinux
-# Clean up after package removal
-if [ $1 -eq 0 ]; then
-  for selinuxvariant in %{selinux_variants}
-    do
-      /usr/sbin/semanage module -s ${selinuxvariant} -l > /dev/null 2>&1 \
-        && /usr/sbin/semodule -s ${selinuxvariant} -r %{modulename} || :
-    done
-fi
-
-rpm -ql osa-dispatcher | xargs -n 1 /sbin/restorecon -rvi {}
-/sbin/restorecon -vvi /var/log/rhn/osa-dispatcher.log
-%endif
-
 %files
 %{_sbindir}/osad
 %config(noreplace) %{_sysconfdir}/sysconfig/rhn/osad.conf
@@ -443,16 +234,6 @@ rpm -ql osa-dispatcher | xargs -n 1 /sbin/restorecon -rvi {}
 %dir %{_sysconfdir}/sysconfig/rhn/clientCaps.d
 %endif
 
-%if 0%{?build_py2}
-%files -n python2-%{name}
-%attr(755,root,root) %{_sbindir}/osad-%{python_version}
-%dir %{python_sitelib}/osad
-%{python_sitelib}/osad/osad.py*
-%{python_sitelib}/osad/osad_client.py*
-%{python_sitelib}/osad/osad_config.py*
-%endif
-
-%if 0%{?build_py3}
 %files -n python3-%{name}
 %attr(755,root,root) %{_sbindir}/osad-%{python3_version}
 %dir %{python3_sitelib}/osad
@@ -463,54 +244,7 @@ rpm -ql osa-dispatcher | xargs -n 1 /sbin/restorecon -rvi {}
 %{python3_sitelib}/osad/__pycache__/osad.*
 %{python3_sitelib}/osad/__pycache__/osad_client.*
 %{python3_sitelib}/osad/__pycache__/osad_config.*
-%endif
 
-%files -n osa-dispatcher
-%{_sbindir}/osa-dispatcher
-%config(noreplace) %{_sysconfdir}/sysconfig/osa-dispatcher
-%config(noreplace) %{_sysconfdir}/logrotate.d/osa-dispatcher
-%{rhnroot}/config-defaults/rhn_osa-dispatcher.conf
-%dir %{_sysconfdir}/rhn/tns_admin
-%dir %{_sysconfdir}/rhn/tns_admin/osa-dispatcher
-%config(noreplace) %{_sysconfdir}/rhn/tns_admin/osa-dispatcher/sqlnet.ora
-%if 0%{?fedora} || 0%{?suse_version} >= 1210 || 0%{?rhel} >= 7
-%{_unitdir}/osa-dispatcher.service
-%else
-%attr(755,root,root) %{_initrddir}/osa-dispatcher
-%endif
-%attr(770,root,%{apache_group}) %dir %{_var}/log/rhn/oracle
-%attr(770,root,root) %dir %{_var}/log/rhn/oracle/osa-dispatcher
-%doc LICENSE
-%ghost %attr(640,%{apache_user},root) %{_var}/log/rhn/osa-dispatcher.log
-%if 0%{?suse_version}
-%attr(750,root,%{apache_group}) %dir %{_sysconfdir}/rhn
-%dir %{rhnroot}
-%attr(755,root,%{apache_group}) %dir %{rhnroot}/config-defaults
-%dir %{_var}/log/rhn
-%endif
-
-%files -n python2-osa-dispatcher
-%attr(755,root,root) %{_sbindir}/osa-dispatcher-%{python_version}
-%dir %{python_sitelib}/osad
-%{python_sitelib}/osad/osa_dispatcher.py*
-%{python_sitelib}/osad/dispatcher_client.py*
-
-%if 0%{?build_py3}
-%files -n python3-osa-dispatcher
-%attr(755,root,root) %{_sbindir}/osa-dispatcher-%{python3_version}
-%dir %{python3_sitelib}/osad
-%{python3_sitelib}/osad/osa_dispatcher.py*
-%{python3_sitelib}/osad/dispatcher_client.py*
-%{python3_sitelib}/osad/__pycache__/osa_dispatcher.*
-%{python3_sitelib}/osad/__pycache__/dispatcher_client.*
-%endif
-
-%files -n python2-osa-common
-%{python_sitelib}/osad/__init__.py*
-%{python_sitelib}/osad/jabber_lib.py*
-%{python_sitelib}/osad/rhn_log.py*
-
-%if 0%{?build_py3}
 %files -n python3-osa-common
 %{python3_sitelib}/osad/__init__.py*
 %{python3_sitelib}/osad/jabber_lib.py*
@@ -518,20 +252,11 @@ rpm -ql osa-dispatcher | xargs -n 1 /sbin/restorecon -rvi {}
 %{python3_sitelib}/osad/__pycache__/__init__.*
 %{python3_sitelib}/osad/__pycache__/jabber_lib.*
 %{python3_sitelib}/osad/__pycache__/rhn_log.*
-%endif
-
-%if 0%{?include_selinux_package}
-%files -n osa-dispatcher-selinux
-%doc osa-dispatcher-selinux/%{modulename}.fc
-%doc osa-dispatcher-selinux/%{modulename}.if
-%doc osa-dispatcher-selinux/%{modulename}.te
-%{_datadir}/selinux/*/%{modulename}.pp
-%{_datadir}/selinux/devel/include/%{moduletype}/%{modulename}.if
-%doc LICENSE
-%attr(0755,root,root) %{_sbindir}/osa-dispatcher-selinux-enable
-%endif
 
 %changelog
+* Mon Apr 16 2018 Tomas Kasparek <tkasparek@redhat.com> 5.11.99-5
+- don't build osa-dispatcher and python2 subpackages (tkasparek@redhat.com)
+
 * Tue Mar 20 2018 Tomas Kasparek <tkasparek@redhat.com> 5.11.99-4
 - remove osad files when packaging only for python3 (tkasparek@redhat.com)
 - osa-dispatcher is dependent on spacewalk-backend which is in python2
